tests/Unit/TracerTest.php

```diff
@@ -269,24 +287,38 @@ public function testHonorGlobalTags()
 
         $this->assertSame('value1', $span->getAllTags()['key1']);
         $this->assertSame('value2', $span->getAllTags()['key2']);
+
+        \putenv('DD_TAGS='); // prevent memory leak
```

Comment:
After clicking "approve" yesterday, I started thinking about this leak a bit more and I wanted to confirm that we don't leak tags by default. The comment above the `HASH` macro, it states:

> This could only ever happens if an user uses \dd_trace_internal_fn('ddtrace_reload_config'); at the wrong point in time - which is an internal testsuite function which users are not supposed to use

But the tracer [invokes this function in the request init hook](https://github.com/DataDog/dd-trace-php/blob/d3bc9f01302a94bc94465507fc61a01968da050d/bridge/dd_init.php#L55). It also runs it from RINIT on [first-time RINIT](https://github.com/DataDog/dd-trace-php/blob/6e1dccff9d69b3e913be764ad661e3a965bd662e/ext/php8/ddtrace.c#L422).

I think that leaves us with a possible memleak at the end of the first request when `DD_TAGS` is non-empty on the happy path, correct?

Reply:
Not correct. It would only result in a memleak if someone accessed the DD_TAGS via `dd_trace_env_config` beforehand without releasing that variable.

Reply:
Got it. In that case, on the second invocation of `ddtrace_reload_config`, the RC == 1 and `zend_hash_release` will free the first allocation that `ddtrace_get_hash_config` made. 👍 Thanks for clarifying.